src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public static final int RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Circle defaultCircle = new Circle(Color.WHITE.name(), RADIUS);
+
+    public Figure getRandomFigure() {
+        int random = new Random().nextInt(5);
+        switch (random) {
+            case 1: { // Triangle
+                double base = getRandom();
+                double height = getRandom();
+                String color = colorSupplier.getRandomColor();
+                return new RightTriangle(color, base, height);
+            }
+            case 2: { // Circle
+                double radius = getRandom();
+                String color = colorSupplier.getRandomColor();
+                return new Circle(color, radius);
+            }
+            case 3: { // Rectangle
+                double length = getRandom();
+                double width = getRandom();
+                String color = colorSupplier.getRandomColor();
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.